gnosis/cowswap
# 1334/activity-orders-from-api

# Summary

Activity orders from API: "Your orders, wherever you are"

Main changes:
- At start/network/account change, fetches orders from API
- Shows only orders for connected account/network
- Shows only txs for connected account/network
- Displays at most 10 regulars orders, and as many pending as there are
- Shows orders sorted by creation date, descending
- Replaced `Clear activity` with `View all orders` ~(Pending CSS work)~
- Added a `View all orders` at the bottom of activity list

# To Test

1. Connect a wallet with existing orders
2. Open Account Activities sidebar
* There should be 10 orders or less, doesn't matter how old they are (There might be more if you have more than 10 pending orders)
* There should be a link on top right that takes you to the Explorer to see all of your orders
![screenshot_2021-09-27_16-24-30](https://user-images.githubusercontent.com/43217/134998360-3c0616ab-59b0-42f5-9d2d-896e3cf663db.png)

3. Change account AND/OR network
* Orders should be updated to match the selected account/network
4. Place an order
5. While it's pending, open the same page in another browser / device (:warning: important to test in a different browser otherwise the same local storage will be used)
* The pending order should be visible as pending in the new browser / device


# *NOT* part of this PR

- Load txs the same way as orders
- Refresh pending orders in the background - right now they are updated only on account/network change
- Show orders that have Tokens not yet loaded in the current UI
